From 4b015c914a3cb702c60cac75b7e704983507941b Mon Sep 17 00:00:00 2001
From: Etaash Mathamsetty
 <45927311+Etaash-mathamsetty@users.noreply.github.com>
Date: Thu, 2 Jul 2026 03:15:04 -0400
Subject: [PATCH 176/285] win32u: Fix is_window_visible/drawable when topmost
 window is the desktop window of another thread.

---
 dlls/win32u/window.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/dlls/win32u/window.c b/dlls/win32u/window.c
index 6afa988f001..de9c59b108d 100644
--- a/dlls/win32u/window.c
+++ b/dlls/win32u/window.c
@@ -957,7 +957,8 @@ BOOL is_window_visible( HWND hwnd )
     {
         for (i = 0; list[i+1]; i++)
             if (!(get_window_long( list[i], GWL_STYLE ) & WS_VISIBLE)) break;
-        retval = !list[i+1] && (list[i] == get_desktop_window());  /* top message window isn't visible */
+        /* top message window isn't visible */
+        retval = !list[i+1];
     }
     free( list );
     return retval;
@@ -986,7 +987,8 @@ BOOL is_window_drawable( HWND hwnd, BOOL icon )
         for (i = 0; list[i+1]; i++)
             if ((get_window_long( list[i], GWL_STYLE ) & (WS_VISIBLE|WS_MINIMIZE)) != WS_VISIBLE)
                 break;
-        retval = !list[i+1] && (list[i] == get_desktop_window());  /* top message window isn't visible */
+        /* top message window isn't visible */
+        retval = !list[i+1];
     }
     free( list );
     return retval;
-- 
2.55.0

